trace-recorder/src/main/org/jetbrains/lincheck/trace/recorder/TraceCollectingEventTracker.kt

```diff
@@ -173,53 +183,97 @@ class TraceCollectingEventTracker(
         }
     }
 
-    override fun beforeThreadFork(thread: Thread, descriptor: ThreadDescriptor) = runInsideIgnoredSection {
+    override fun beforeExistingThreadTracking(thread: Thread, descriptor: ThreadDescriptor): Unit = runInsideIgnoredSection {
+        val threadData = threads.computeIfAbsent(thread) {
+            val threadData = ThreadData(threads.size)
+            ThreadDescriptor.getThreadDescriptor(thread).eventTrackerData = threadData
+            threadData
+        }
+        val thread = Thread.currentThread()
+        descriptor.enableAnalysis()
+
+        fun appendMethodCall(obj: TRObject?, className: String, methodName: String, methodDesc: String, codeLocationId: Int = -1, params: List<TRObject> = emptyList()) {
+            val methodCall = TRIncompleteMethodCallTracePoint(
+                threadId = threadData.threadId,
+                codeLocationId = codeLocationId,
+                methodId = TRACE_CONTEXT.getOrCreateMethodId(className, methodName, methodDesc),
+                obj = obj,
+                parameters = params
+            )
+            val parentTracePoint = if (threadData.getStack().isEmpty()) null
+                                   else threadData.currentMethodCallTracePoint()
+            strategy.tracePointCreated(parentTracePoint, methodCall)
+            threadData.pushStackFrame(methodCall, thread, isInline = false)
+        }
+
+        runInsideInjectedCode {
```

Comment:
Why the whole method is not enclosed in `runInsideInjectedCode` ? 
Could you please add a comment?

Reply:
`runInsideInjectedCode` internally checks that `isAnalysisEnabled == true` (when performing a double-checking), so the `descriptor.enableAnalysis()` should be called before the `runInsideInjectedCode`.

And that is also the reason why `runInsideInjectedCode` does not wrap the whole method: analysis of this thread has not been enabled yet, so `runInsideInjectedCode` would just exit without even invoking the body of this method